fabric-convention-tags-v2/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/BlockTagGenerator.java

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.tag.convention.datagen.generators;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.registry.RegistryWrapper;
+import net.minecraft.registry.tag.BlockTags;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.datagen.v1.FabricDataOutput;
+import net.fabricmc.fabric.api.datagen.v1.provider.FabricTagProvider;
+import net.fabricmc.fabric.api.tag.convention.v2.ConventionalBlockTags;
+import net.fabricmc.fabric.api.tag.convention.v2.TagUtil;
+
+public final class BlockTagGenerator extends FabricTagProvider.BlockTagProvider {
+	static List<Block> VILLAGER_JOB_SITE_BLOCKS = List.of(
+			Blocks.BARREL,
+			Blocks.BLAST_FURNACE,
+			Blocks.BREWING_STAND,
+			Blocks.CARTOGRAPHY_TABLE,
+			Blocks.CAULDRON,
+			Blocks.LAVA_CAULDRON,
+			Blocks.WATER_CAULDRON,
+			Blocks.POWDER_SNOW_CAULDRON,
+			Blocks.COMPOSTER,
+			Blocks.FLETCHING_TABLE,
+			Blocks.GRINDSTONE,
+			Blocks.LECTERN,
+			Blocks.LOOM,
+			Blocks.SMITHING_TABLE,
+			Blocks.SMOKER,
+			Blocks.STONECUTTER
+	);
+
+	public BlockTagGenerator(FabricDataOutput output, CompletableFuture<RegistryWrapper.WrapperLookup> registriesFuture) {
+		super(output, registriesFuture);
+	}
+
+	@Override
+	protected void configure(RegistryWrapper.WrapperLookup registries) {
+		getOrCreateTagBuilder(ConventionalBlockTags.STONES)
+				.add(Blocks.ANDESITE)
+				.add(Blocks.DIORITE)
+				.add(Blocks.GRANITE)
+				.add(Blocks.INFESTED_STONE)
+				.add(Blocks.STONE)
+				.add(Blocks.DEEPSLATE)
+				.add(Blocks.INFESTED_DEEPSLATE)
+				.add(Blocks.TUFF);
+
+		getOrCreateTagBuilder(ConventionalBlockTags.QUARTZ_ORES)
+				.add(Blocks.NETHER_QUARTZ_ORE);
+		getOrCreateTagBuilder(ConventionalBlockTags.NETHERITE_SCRAP_ORES)
+				.add(Blocks.ANCIENT_DEBRIS);
+		getOrCreateTagBuilder(ConventionalBlockTags.ORES)
+				.addOptionalTag(BlockTags.REDSTONE_ORES)
```

Comment:
Ore ordering here is odd. I'd opt for metals -> gems -> quartz -> netherite but really any solid ordering is fine.